src/declaration.ts

```diff
@@ -37,6 +37,31 @@ export class FunctionDecl<F extends AnyFunction = AnyFunction> extends BaseDecl<
   }
 }
 
+export const isNativeFunctionDecl = typeGuard("NativeFunctionDecl");
+
+/**
+ * A function declaration which contains the original closure instead of Functionless expressions.
+ */
+export class NativeFunctionDecl<
+  F extends AnyFunction = AnyFunction
+> extends BaseDecl<"NativeFunctionDecl", undefined> {
+  readonly _functionBrand?: F;
+  constructor(
+    readonly parameters: ParameterDecl[],
```

Comment:
Why don't we also include the `FunctionDecl`? We can use it to auto-wire IAM permissions and environment variables.

Reply:
I had a feeling of need it eventually, but wasn't sure. This is probably where we'd end up.

Reply:
Current implementation extracts the integrations during the native expr parsing. 

I question the need to output the entire `FunctionDecl` and risking any compilation failures when we don't make use of most of the output. 

Going to leave this as is for now.